src/tools/tidy/src/deps.rs

```diff
@@ -171,6 +171,7 @@ const WHITELIST: &[Crate<'_>] = &[
     Crate("vcpkg"),
     Crate("version_check"),
     Crate("void"),
+    Crate("wasi"),
```

Comment:
I've opened an issue on the wasi crate repo about its license: https://github.com/CraneStation/rust-wasi/issues/9

Reply:
Yeah, it's not part of std (it's used only for tests and in rustc) so the bar is lower here but still it'd be nice to have more deps licensed the same way as Rust.

Reply:
Note that if #62082 or #63676 will merge, `wasi` will be a part of `std`.

Reply:
The next wasi version will be licensed under Apache2 OR MIT, so this has been resolved.